airbyte-integrations/connectors/source-shopify/source_shopify/streams/base_streams.py

```diff
@@ -0,0 +1,684 @@
+        https://shopify.dev/docs/api/usage/bulk-operations/queries#write-a-bulk-operation
+        """
+        return None
+
+    def request_body_json(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> Mapping[str, Any]:
+        """
+        Override for _send_request CDK method to send HTTP request to Shopify BULK Operatoions.
+        https://shopify.dev/docs/api/usage/bulk-operations/queries#bulk-query-overview
+        """
+        return {"query": ShopifyBulkTemplates.prepare(stream_slice.get("query"))}
+
+    def _send_request(self, request: requests.PreparedRequest, request_kwargs: Mapping[str, Any]) -> Optional[requests.Response]:
+        """
+        Override for _send_request CDK method to send HTTP request to Shopify BULK Operatoions.
+        https://shopify.dev/docs/api/usage/bulk-operations/queries#bulk-query-overview
+        """
+        # Create the BULK Job
+        return self.bulk_job.job_create(request)
+
+    @stream_state_cache.cache_stream_state
+    def stream_slices(self, stream_state: Optional[Mapping[str, Any]] = None, **kwargs) -> Iterable[Optional[Mapping[str, Any]]]:
+        if self.filter_field:
+            start = pdm.parse(stream_state.get(self.cursor_field) if stream_state else self.config.get("start_date"))
+            end = pdm.now()
+            while start < end:
+                # check if we reached the max attempts retry with concurent BULK job.
+                if self.bulk_job.concurrent_max_attempt_reached:
+                    # exit gracefully from the sync
+                    return []
+                else:
+                    slice_end = start.add(days=self.slice_interval_in_days)
+                    # check end period is less than now() or now() is applied otherwise.
+                    slice_end = slice_end if slice_end < end else end
+                    # making pre-defined sliced query to pass it directly
+                    prepared_query = self.query.get(self.filter_field, start.to_rfc3339_string(), slice_end.to_rfc3339_string())
+                    self.logger.info(f"Stream: `{self.name}` requesting BULK Job for period: {start} -- {slice_end}.")
+                    yield {"query": prepared_query}
+                    start = slice_end
+        else:
+            # for the streams that don't support filtering
+            yield {"query": self.query.get()}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        # get the cached substream state, to avoid state collisions for Incremental Syncs
+        stream_state = stream_state_cache.cached_state.get(self.name, {self.cursor_field: self.config["start_date"]})
+        # get job_result from COMPLETED BULK Job
+        job_result_url = self.bulk_job.job_check(self.graphql_path, response)
+        # the `job_result_url` could be `None`,
+        # meaning there are no data available for the slice period.
+        if job_result_url:
```

Comment:
it seems like we'll silently ignore the response if we couldn't extract a job_id. shouldn't we fail if that happens or are there valid cases? If there are valid cases, we should probably at least log something so we know what's going on when troubleshooting issues

Reply:
The cases when it happens:
1) when we have  the  `concurrent` job running 
2) when the created job has nothing to return (data == null) for the requested date period, and it instantly gives `COMPLETED`  status, with no `result_url` - typically an `empty` result from the slice.

I'd prefer to keep it silent.